File: pegasusX/apps/driver-app-android/app/src/main/java/com/pegasusx/driver/service/EnterpriseUpdateConfig.kt
package com.pegasusx.driver.service

import com.pegasusx.driver.BuildConfig

/**
 * Distribution config for native Android.
 *
 * - **enterprise** flavor: website CDN OTA (`channel=enterprise`, ENABLE_CDN_OTA=true)
 * - **store** flavor: Play Store (`channel=production`, ENABLE_CDN_OTA=false)
 *
 * Build with: `./gradlew :app:assembleStoreRelease` or `assembleEnterpriseRelease`.
 */
object EnterpriseUpdateConfig {
    /** Client-policy channel — production for Play, enterprise for website. */
    val CHANNEL: String
        get() = BuildConfig.DISTRIBUTION_CHANNEL

    /** When false, never download/install APK from CDN; open Play listing instead. */
    val enableCdnOta: Boolean
        get() = BuildConfig.ENABLE_CDN_OTA

    /** Policy role for client-policy API. */
    const val POLICY_ROLE = "DRIVER"

    /**
     * Fallback CDN manifest for enterprise only.
     * Store builds ignore this and use [storeListingUrl].
     */
    const val DEFAULT_MANIFEST_URL =
        "https://storage.googleapis.com/pegasusx-ssmr-app-updates/android/driver/updater.json"

    /** Play Store listing (store flavor); empty on enterprise flavor. */
    val storeListingUrl: String
        get() = BuildConfig.STORE_LISTING_URL.ifBlank {
            "https://play.google.com/store/apps/details?id=${BuildConfig.APPLICATION_ID}"
        }

    val isStoreBuild: Boolean
        get() = !enableCdnOta
}

File: pegasusX/apps/driver-app-ios/driverapp/driverapp/Services/EnterpriseUpdateConfig.swift
import Foundation

/// Legacy mirror — prefer driverappios tree for shipping builds.
enum EnterpriseUpdateConfig {
    static var channel: String {
        if let v = Bundle.main.object(forInfoDictionaryKey: "PXDistributionChannel") as? String,
           !v.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty {
            return v.trimmingCharacters(in: .whitespacesAndNewlines).lowercased()
        }
        #if STORE_BUILD
        return "production"
        #else
        return "enterprise"
        #endif
    }

    static var enableCdnOta: Bool {
        if let v = Bundle.main.object(forInfoDictionaryKey: "PXEnableCdnOta") as? Bool { return v }
        #if STORE_BUILD
        return false
        #else
        return channel != "production" && channel != "store"
        #endif
    }

    static let policyRole = "DRIVER"

    static let defaultManifestURL = URL(
        string: "https://storage.googleapis.com/pegasusx-ssmr-app-updates/ios/driver/updater.json"
    )!

    static var storeListingURL: URL? {
        if let s = Bundle.main.object(forInfoDictionaryKey: "PXAppStoreURL") as? String,
           let u = URL(string: s), !s.isEmpty { return u }
        if let id = Bundle.main.object(forInfoDictionaryKey: "PXAppStoreID") as? String {
            let trimmed = id.trimmingCharacters(in: .whitespacesAndNewlines)
                .replacingOccurrences(of: "id", with: "", options: [.anchored])
            if !trimmed.isEmpty {
                return URL(string: "https://apps.apple.com/app/id\(trimmed)")
            }
        }
        return nil
    }
}
